src/charm.py

```diff
@@ -48,13 +49,15 @@ def __init__(self, *args: typing.Any) -> None:
         """
         super().__init__(*args)
         self._backup = BackupObserver(self)
+        self._irc_bridge_database = DatabaseObserver(self, relation_name="irc-bridge-database")
```

Comment:
Out of curiosity: Is there a reason for this not to be set only if enable_irc_bridge is true? Together with self._irc_bridge.

Reply:
Nope, it should be within the condition, thanks!